### PR TITLE
modify_yaml: handle None value during update.

### PR DESCRIPTION
**Pull request by abutcher**

### CODE REVIEW

File: library/modify_yaml.py [hunk 1]
```diff
@@ -25,15 +25,22 @@
 def set_key(yaml_data, yaml_key, yaml_value):
     changes = []
     ptr = yaml_data
+    final_key = yaml_key.split('.')[-1]
```
Comment (tbielawa) [inline library/modify_yaml.py]: great addition. makes this 100% easier to read instantly.

File: library/modify_yaml.py [hunk 2]
```diff
@@ -25,15 +25,22 @@
...
     for key in yaml_key.split('.'):
-        if key not in ptr and key != yaml_key.split('.')[-1]:
+        # Key isn't present and we're not on the final key. Set to empty dictionary.
```
Comment (tbielawa) [inline library/modify_yaml.py]: # I love comments

File: library/modify_yaml.py [hunk 3]
```diff
@@ -71,21 +78,18 @@ def none_representer(dumper, data):
     yaml.add_representer(type(None), none_representer)
 
     try:
-
-        yaml_file = open(dest)
-        yaml_data = yaml.safe_load(yaml_file.read())
-        yaml_file.close()
+        with open(dest) as yaml_file:
```
Comment (tbielawa) [inline library/modify_yaml.py]: smart improvement

**Comment by sdodson** [comment]
aos-ci-test

**Pull request merged**